File: src/main/java/pw/aru/commands/funny/ShadowDice.java
package pw.aru.commands.funny;

import java.util.concurrent.ThreadLocalRandom;

public class ShadowDice {

    private int clamp(int v, int min, int max) {
        return Math.min(Math.max(v, min), max);
    }

    public int roll(int sides) {
        ThreadLocalRandom random = ThreadLocalRandom.current();

        return clamp((int) ((random.nextDouble() * 0.75 + random.nextGaussian() * 0.25) * sides), 0, sides - 1) + 1;
    }
}

File: src/main/kotlin/pw/aru/commands/funny/ShadowDice.kt
package pw.aru.commands.funny

import java.util.concurrent.ThreadLocalRandom
import kotlin.math.max
import kotlin.math.min

class ShadowDice {
    private fun Int.clamp(min: Int, max: Int) = min(max(this, min), max)

    fun roll(sides: Int): Int {
        val random = ThreadLocalRandom.current()

        return ((random.nextDouble() * 0.75 + random.nextGaussian() * 0.25) * sides).toInt().clamp(0, sides - 1) + 1
    }
}
